.github/workflows/test-pr.yml

```diff
@@ -21,27 +21,27 @@ jobs:
       - name: 'Set up Docker'
         uses: ./.github/actions/with-docker
         with:
-          tag: kevm-ci-concrete
+          tag: kevm-ci-concrete-${GITHUB_SHA}
       - name: 'Build kevm-pyk'
-        run: docker exec -t kevm-ci-concrete /bin/bash -c 'make kevm-pyk venv -j`nproc`'
+        run: docker exec -t kevm-ci-concrete-${GITHUB_SHA} /bin/bash -c 'make kevm-pyk venv -j`nproc`'
       - name: 'Build kevm'
```

Comment:
What abotu the other steps that use docker containers? The java and haskell and foundry steps?